cmake/toolchain.cmake

```diff
@@ -18,7 +18,7 @@ set(CMAKE_C_STANDARD 11)
 if(NOT WIN32)
     # Compiler options are set only on non-Windows, since these options
     # are not correct for MSVC.
-    set(CMAKE_C_FLAGS_INIT "-Wall")
+    set(CMAKE_C_FLAGS_INIT "-Wall -Werror")
     string(CONCAT CMAKE_C_FLAGS_DEBUG_INIT
            "-g -gdwarf-2 -g3 -O0 -fno-inline -fno-eliminate-unused-debug-types "
            "--coverage")
```

Comment:
At least a few of these were added by @dfellis in #8 (1bfa8631). Not sure about some of the others.

Reply:
OK thanks. At some point it might be worth revisiting the flags here. I am not sure if they are the best options and if there is a better way to use the toolchain file.